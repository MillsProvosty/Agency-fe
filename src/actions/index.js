export const setRole = role => ({
  type: 'SET_ROLE',
  role
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setUserOpportunities = opportunities => ({
  type: 'SET_USER_OPPORTUNITIES',
  opportunities
});

export const addOpp = opp => ({
  type: 'ADD_OPP',
  opp
});

export const editOpp = opp => ({
  type: 'EDIT_OPP',
  opp
});
