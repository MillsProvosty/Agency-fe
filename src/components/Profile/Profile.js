import React, { useState } from "react";
import "./Profile.scss";
import Nav from "../Nav/Nav";
import Modal from "react-modal";

const Profile = () => {
  const user = {
    firstName: "Confetti",
    lastName: "Hornblower",
    role: "client",
    email: ""
  };

  const [createModal, showCreateModal] = useState(false);
  const [searchModal, showSearchModal] = useState(false);

  return (
    <section className="Profile">
      <Nav />
      <section className="container-dash">
        <Modal isOpen={createModal}>
          <CreateOppModal />
        </Modal>
        <Modal isOpen={searchModal}>
          <CreateOppModal />
        </Modal>
        <h1>Welcome {user.firstName}</h1>
        {user.role === "client" && <button onClick={() => showCreateModal(true)}>Search for opportunities</button>}
        {user.role === "volunteer" && <button onClick={() => showSearchModal(true)}>Make another opportunity</button>}
      </section>
    </section>
  );
};

export default Profile;
