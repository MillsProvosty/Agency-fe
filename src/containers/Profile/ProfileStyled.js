import styled from "styled-components";

export const ProfileSection = styled.section`
  width: 100%;
  height: 100vh;
  background-color: white;
  background-attachment: fixed;
  @media screen and (max-width: 375px) {
    justify-content: space-around;
  }
`;

export const Header = styled.h1`
color: #37474E
font-size: 2.5em;
margin: 1em;
@media screen and (max-width: 375px) {
  margin: 0px 0px 30px -80px
}
`;
