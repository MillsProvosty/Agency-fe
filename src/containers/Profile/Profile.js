import React, { useState, useEffect } from "react";
import "./Profile.scss";
import Nav from "../../components/Nav/Nav";
import { connect } from "react-redux";
import { getAllOpportunities, deleteAUser } from "../../util/apiCalls";
import { setUserOpportunities } from "../../actions";
import Opportunities from "../Opportunities/Opportunities";
import styled from "styled-components";

const ProfileSection = styled.section`
  width: 100%;
  height: 100vh;
  background-color: aliceblue;
  background-attachment: fixed;
  @media screen and (max-width: 375px) {
    justify-content: space-around;
  }
`;

const Header = styled.h1`
color: #37474E
font-size: 6em;
margin: 0px;
@media screen and (max-width: 375px) {
  margin: 0px 0px 30px -80px
}
`;

const Profile = props => {
  const [isLoading, setLoading] = useState(true);

const { user } = props
  const getUserOpp = async () => {
    setLoading(false);
  };

  useEffect(() => {
    getUserOpp();
  }, []);

  return (
    <ProfileSection className="Profile">
      <Nav />
      <button onClick={() => deleteAUser(user.id)}>Delete Account</button>
      {isLoading && <p>I am loading</p>}
      {!isLoading && user.role === "client" && (
        <>
          <Header>Welcome, {user.firstname}</Header>
          <Opportunities role={user.role} />
        </>
      )}
      {!isLoading && user.role === "volunteer" && (
        <>
          <h1>Welcome {user.firstname}</h1>
          <Opportunities role={user.role} />
        </>
      )}
    </ProfileSection>
  );
};

export const mapStateToProps = state => ({
  user: state.user,
  opportunities: state.opportunities
});

export const mapDispatchToProps = dispatch => ({
  setOpportunities: opportunities =>
    dispatch(setUserOpportunities(opportunities))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Profile);
