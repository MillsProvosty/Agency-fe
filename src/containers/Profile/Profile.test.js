import React from 'react';
import { shallow } from 'enzyme';
import { Profile, mapStateToProps, mapDispatchToProps } from './Profile';
import { setAllOpportunities } from '../../actions'

describe('Profile', () => {
  let wrapper;


  const mockUser = {
    role: 'client',
    first_name: 'Bob',
    id: 1
  }

  const mockOpp = [{
    "title": "aint no thang but a chicken wang",
    "type": "gangsta",
    "location": "atlanta ga",
    "estimated_time": "1",
    "description": "357 to yo forehead" 
  }]

  const mockSetOpp = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <Profile 
        user={mockUser}
        opportunities={mockOpp}
        setOpportunities={mockSetOpp}
      />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapStateToProps',() => {
    it('should return an array of opportunities', () => {
      let mockState = {
        user: {
          id: 1,
          name: 'something'
        },
        opportunities: mockOpp,
        role: 'client'
      }
      let expected = mockOpp
  
      let mappedProps = mapStateToProps(mockState);
  
      expect(mappedProps.opportunities).toEqual(expected)
    })

    it('should return a user', () => {
      let mockState = {
        user: {
          id: 1,
          name: 'something'
        },
        opportunities: mockOpp,
        role: 'client'
      }
      let expected = mockState.user
      let mappedProps = mapStateToProps(mockState);
  
      expect(mappedProps.user).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it("calls setOpportnities with an setAllOpportunities action when setOpportunities is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setAllOpportunities(mockOpp);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setOpportunities(mockOpp);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})
