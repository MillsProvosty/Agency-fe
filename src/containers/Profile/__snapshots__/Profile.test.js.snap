// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match snapshot 1`] = `
<styled.section
  className="Profile"
>
  <Nav />
  <button
    onClick={[Function]}
  >
    Delete Account
  </button>
  <p>
    I am loading
  </p>
</styled.section>
`;
